client/src/main/scala/org/http4s/client/PoolManager.scala

```diff
@@ -1,32 +1,41 @@
 package org.http4s
 package client
 
+import java.time.Instant
+import java.util.concurrent.TimeoutException
+
 import cats.effect._
 import fs2.async
 import org.log4s.getLogger
 
 import scala.annotation.tailrec
 import scala.collection.mutable
 import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
 import scala.util.Random
 
 private final class PoolManager[F[_], A <: Connection[F]](
     builder: ConnectionBuilder[F, A],
     maxTotal: Int,
     maxWaitQueueLimit: Int,
     maxConnectionsPerRequestKey: RequestKey => Int,
+    responseHeaderTimeout: Duration,
```

Comment:
If you are doing an `isFinite` check with this every time, is there a reason this can't be a `FiniteDuration`?

Reply:
We need to support no timeouts (infinite timeouts).

Reply:
oh herp derp. `Duration.Inf` is a thing. Forgot. My bad :+1: